AsyncDisplayKit/ASCollectionNode.mm

```diff
@@ -273,6 +317,20 @@ - (void)setTuningParameters:(ASRangeTuningParameters)tuningParameters forRangeMo
   return [self.rangeController setTuningParameters:tuningParameters forRangeMode:rangeMode rangeType:rangeType];
 }
 
+#pragma mark - Selection
+
+- (void)selectItemAtIndexPath:(nullable NSIndexPath *)indexPath animated:(BOOL)animated scrollPosition:(UICollectionViewScrollPosition)scrollPosition
+{
+  ASDisplayNodeAssert([self isNodeLoaded], @"ASCollectionNode should be loaded before calling selectItemAtIndexPath");
```

Comment:
In the future it will be nice to remove this restriction. It would require that we have a clean method for ensuring that the view is loaded, that the initial reload has been called, and that all updates are committed. Today we could cobble together all that into a method but it would be ugly and there are other potential hazards (what if we're inside `willDisplayCell:forItemAtIndexPath:` and waiting is unsafe, what if we're zero-size, etc.) That will come later and I think that this is plenty for the medium-term.

Also, would you be willing to add `ASDisplayNodeAssertMainThread()` in these methods?